app/gui2/src/components/GraphEditor/GraphNode.vue

```diff
@@ -143,6 +162,70 @@ const dragPointer = usePointer((pos, event, type) => {
   }
 })
 
+const exprWithoutOverriddenOutputContext = computed(() => {
+  const matches = extractMatches(props.node.rootSpan, withOverriddenOutputContextPatternAst)
+  if (!matches) return
+  let enabled: boolean
+  switch (matches[0]?.repr()) {
+    case enableMethodName: {
+      enabled = true
+      break
+    }
+    case disableMethodName: {
+      enabled = false
+      break
+    }
+    default: {
+      return
+    }
+  }
+  return { enabled, contextExpr: matches[1]!, innerExpr: matches[2]! }
+})
+
+const displayedExpression = computed(
+  () => exprWithoutOverriddenOutputContext.value?.innerExpr ?? props.node.rootSpan,
+)
+
+const isOutputContextOverridden = computed({
+  get() {
+    return (
+      exprWithoutOverriddenOutputContext.value != null &&
+      exprWithoutOverriddenOutputContext.value.enabled !== projectStore.isOutputContextEnabled &&
+      exprWithoutOverriddenOutputContext.value.contextExpr.repr().slice(1, -1) ===
+        projectStore.executionMode
```

Comment:
I would make an alias for `exprWithoutOverriddenOutputContext.value` to shorten this expression somewhat. Also, the last could be a bit decomposed; the value returned from `slice` should be a semantically named variable for easy reading.